### ISSUE TITLE
Activation Bytes decryption service

### ISSUE THREAD

**Issue by JKamsker** [openaudible/openaudible#422]
Ive created a service which decrypts any hash into the correct activationbytes.
You can use it if you want.

Here is how the webapi is used: https://github.com/JKamsker/Aax.Activation.ApiClient/blob/master/Aax.Activation.ApiClient/AaxActivationClient.cs

The service url is '"https://aaxactivationserviceapi.azurewebsites.net/api/v1/activation/" + you checksum'.
Please remember cache the results

**Comment by Strider3000** [comment]
ELI5 what this does

**Comment by JKamsker** [comment]
Open "https://aaxactivationserviceapi.azurewebsites.net/api/v1/activation/" + you checksum in your browser and it will show you the activatonbytes @Strider3000

**Comment by Strider3000** [comment]
@JKamsker Ohhhh cool do you have a massive precompiled rainbowtable?
You might want to include a brief readme. 
Example:
1. Get hash:

ffprobe Agamemnon.AAX

2. Get activation bytes:

lynx -dump -nolist https://aaxactivationserviceapi.azurewebsites.net/api/v1/activation/b246810123450a9976be4a215783b6f8ad86506b

3. Convert:

ffmpeg -i Agamemnon.AAX -activation_bytes 5e17f81c -vn -c:a copy Agamemnon.mp4

**Comment by JKamsker** [comment]
@Strider3000 I wouldnt call an optimized db of 15gb "massive" :D

What do you mean by "rename"?

 I mean, it should be incorporated into openaudible anyways... The tutorial wouldnt be necessary that way.

**Comment by Strider3000** [comment]
? @JKamsker I never said rename?

**Comment by JKamsker** [comment]
Oh missread xD
Readme not rename xD

Sooo ok why would one need a readme for just one api endpoint?

**Comment by Strider3000** [comment]
So silly americans don't bug you with questions on github

**Comment by feldegast** [comment]
there is also a cli app on github that does this too audible-activator

i used it with AAXtoMP3 (also in github) to make m4b files from aax files

**Comment by feldegast** [comment]
strider, for good documentation, so there is no ambiguity as to what parameters to give it etc

**Comment by JKamsker** [comment]
@feldegast True, but audible-activator needs a complex installation and for me the usage didnt seem straight forward.

This api just returns the activation bytes, no questions asked. No username/password, no python and whats most important: no connection to the audible servers which probably log and detect those malitious requests.

**Comment by feldegast** [comment]
oh! it doesn't connect to amazon! in that case i totally see a point for this! great work! my only question is, where do you get the checksum from? that's what I'd put in the readme, how to get the checksum and what to do with it once you have it

**Comment by JKamsker** [comment]
@feldegast To explain it in detail, a readme wouldnt be enougth.
All i can say right now, its a specialized dictionary-like database where i do the lookup on.

**Comment by JKamsker** [comment]
@openaudible any plans for using it in the app?

**Comment by openaudible** [comment]
Hi @JKamsker 

I appreciate the effort and thought. 

I want the app self-contained and operated by the user.

With a service like yours, I'd be worried about privacy, reliability, longevity, down-time, DOS attacks, and compensation.  

When lookup fails on a user's end, it usually means ffmpeg is also going to also fail, which just moves the point of failure. 

Really appreciate the offer.

**Comment by JKamsker** [comment]
Hey @openaudible,

thanks for your answers. I understand the concern. 

1: Privacy: What exactly are you concerned about?
2: Reliability: Its hosted on azure, so i can scale it out any time worldwide if needed.
3: Longevity: Cost is next to nothing and azure has a really good uptime :)
4: Dos attacks: Azure.....
5: Compensation: Dont worry about that, it doesnt cost alot. Its not even a 0,1€ a month and i get 50€ free each month :)
6: Lookup failure:
    - If the request returns a false result, you can check it with the given algorithm: https://github.com/inAudible-NG/tables/blob/master/alglib1.cpp#L37
    - You can make it completely optional to the user wether he wants to use the service or not.

**Comment by openaudible** [comment]
I try to give as much privacy to my users. The fewer network calls the better.

DOS is a real issue-- If I send a thousand concurrent requests to your endpoint with a random hash, your service will either die, fail real requests, or if its using Azure Automation, it will rack up big bills. 

Thank you again for the offer!

**Comment by JKamsker** [comment]
@openaudible Hmm...yea sure. But thats why it could be optional.

About the DOS: Can't we just try it out and decide wether it should be continued or not? If there will be a problem, we could just disable this feature with an update.

Thank you for your patience with me 😅
